server/routes/article.js

```diff
@@ -60,7 +65,16 @@ articleRouter.get(
  * @description - Route gets all published articles
  * @returns - It returns an array of all published articles
  */
-articleRouter.get("/", validatePaginationParameters, getAllArticles);
+articleRouter.get("/",
+  validatePaginationParameters,
+  (req, res, next) => {
+    if ('search' in req.query) {
+      next();
+    } else {
+      getAllArticles(req, res, next);
+    }
+  }
+);
 
```

Comment:
i don't really understand what is going on here, this convention is kind of like strange to this project

Reply:
Please can you make it a bit more specific what you find wrong about this